src/transformers/models/gemma/modeling_gemma.py

```diff
@@ -1084,6 +1114,10 @@ def forward(
             cache_position=cache_position,
         )
 
+        if self.config.cache_implementation == "static":
+            # Outputs should not include `Cache`` object as `torch.export`` does not support `Cache` type as an output.
+            outputs.past_key_values = None
```

Comment:
@ArthurZucker The reason of unsetting `past_key_values` in the `outputs` is due to the restriction in `torch.export()`. Neither the inputs nor outputs can have `Cache` object.
```
E    torch._dynamo.exc.UserError: It looks like one of the outputs with type `<class 'transformers.cache_utils.StaticCache'>` is not supported or pytree-flattenable.
E    Exported graphs inputs can only contain the following supported types: [<class 'torch.Tensor'>, <class 'torch.SymInt'>, <class 'torch.SymFloat'>, <class 'torch.SymBool'>, <class 'torch.ScriptObject'>, <class 'NoneType'>, <class 'complex'>, <class 'torch.dtype'>, <class 'str'>, <class 'bool'>, <class 'ellipsis'>, <class 'int'>, <class 'torch.layout'>, <class 'code'>, <class 'torch.memory_format'>, <class 'bytes'>, <class 'float'>, <class 'torch.device'>].
```

We should only unset the `outputs.past_key_values` when the model is used for export. What's your suggestion to move forward? Maybe we could have another field `set_exportable` in the `cache_config` and only set it when it's used for `torch.export` and hide unexportable params from I/O?

Reply:
Perhaps we could create a class that wraps any `ForCausalLM` model, and removes any incompatible output from `torch.export`'s perspective!

On our end, it's very important to avoid making changes to the modeling code itself (i.e. everything in `modeling_xxx.py`). These files are used in many places, including outside the HF ecosystem, so we need to be out of options before making changes there 🤗